deploy/templates/bicep/ContainerAppSetup.bicep

```diff
@@ -0,0 +1,385 @@
+          env: [
+            {
+              name: 'APPINSIGHTS_INSTRUMENTATIONKEY'
+              value: appInsights.properties.InstrumentationKey
+            }
+            {
+              name: 'EventBatching__FlushTimespan'
+              value: '30'
+            }
+            {
+              name: 'EventBatching__MaxEvents'
+              value: '20'
+            }
+            {
+              name: 'Checkpoint__BlobPrefix'
+              value: 'Normalization'
+            }
+            {
+              name: 'CheckpointStorage__AuthenticationType'
+              value: 'ManagedIdentity'
+            }
+            {
+              name: 'CheckpointStorage__BlobStorageContainerUri'
+              value: '${storageAccount.properties.primaryEndpoints.blob}checkpoint'
+            }
+            {
+              name: 'CheckpointStorage__BlobContainerName'
+              value: 'checkpoint'
+            } 
+            {
+              name: 'TemplateStorage__AuthenticationType'
+              value: 'ManagedIdentity'
+            } 
+            {
+              name: 'TemplateStorage__BlobStorageContainerUri'
+              value: '${storageAccount.properties.primaryEndpoints.blob}template'
+            }
+            {
+              name: 'TemplateStorage__BlobContainerName'
+              value: 'template'
+            }
+            {
+              name: 'InputEventHub__AuthenticationType'
+              value: 'ManagedIdentity'
+            }
+            {
+              name: 'InputEventHub__EventHubNamespaceFQDN'
+              value: eventhubNamespace.properties.serviceBusEndpoint
+            }
+            {
+              name: 'InputEventHub__EventHubConsumerGroup'
+              value: '$Default'
+            }
+            {
+              name: 'InputEventHub__EventHubName'
+              value: 'devicedata'
+            }
+            {
+              name: 'NormalizationEventHub__AuthenticationType'
+              value: 'ManagedIdentity'
+            }
+            {
+              name: 'NormalizationEventHub__EventHubNamespaceFQDN'
+              value: eventhubNamespace.properties.serviceBusEndpoint
+            }
+            {
+              name: 'NormalizationEventHub__EventHubConsumerGroup'
+              value: '$Default'
+            }
+            {
+              name: 'NormalizationEventHub__EventHubName'
+              value: 'normalizeddata'
+            } 
+            {
+              name: 'Template__DeviceContent'
+              value: 'devicecontent.json'
+            }
+          ]
+        }
+      ]
+      scale: {
+        minReplicas: 1
+        maxReplicas: 10
+      }
+    }
+  }
+}
+
+resource fhirTransformationContainerApp 'Microsoft.App/containerApps@2022-03-01' ={
+  name: 'fhir-transformation'
+  location: location
+  tags: {
+    IomtFhirConnector: 'ResourceIdentity:Create'
+    IomtFhirVersion: 'R4'
+  }
+  identity: {
+    type: 'SystemAssigned,UserAssigned'
```

Comment:
How is SystemAssigned and UserAssigned being used?  I would think it should only be 1.

Reply:
The UserAssigned identity is being used to pull the container image from the ACR. The same UserAssigned identity is being used to pull the image for the Normalization and FHIR transformation Container Apps. Each Container App also has its own SystemAssigned Identity to assign permissions to other resources that are used by each service. A UserAssigned identity needs to be used for pulling the image since the SystemAssigned Identity isn't created until after the Container App is deployed. I could create separate UserAssigned identities for Normalization and FHIR Transformation to encompass all the necessary permissions (AcrPull + permissions specific to Normalization/FHIR Transformation). However, this UserAssigned identity is also being used to run the script in BuildContainerImages.bicep which pushes the images to ACR so a separate UserIdentity would need to be created as well with AcrPush + Contributor permissions. I am happy to make these changes if you think the separation would make more sense.